packages/web3-eth-contract/test/unit/contract.test.ts

```diff
@@ -66,4 +72,406 @@ describe('Contract', () => {
+
+			const contract = new Contract(GreeterAbi);
+
+			const spyTx = jest
+				.spyOn(eth, 'sendTransaction')
+				.mockImplementation((_objInstance, _tx) => {
+					const newContract = contract.clone();
+					newContract.options.address = deployedAddr;
+					// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+					return Promise.resolve(newContract) as any;
+				});
+
+			const spyEthCall = jest.spyOn(eth, 'call').mockImplementation((_objInstance, _tx) => {
+				expect(_tx.to).toStrictEqual(deployedAddr);
+				expect(_tx.data).toBe('0xcfae3217');
+				// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+				return Promise.resolve(encodedArg) as any; // contract class should decode encodedArg
+			});
+
+			const deployedContract = await contract
+				.deploy({
+					data: GreeterBytecode,
+					arguments: ['My Greeting'],
+				})
+				.send(sendOptions);
+
+			const res = await deployedContract.methods.greet().call();
+			expect(res).toStrictEqual(arg);
+
+			spyTx.mockClear();
+			spyEthCall.mockClear();
+		});
+
+		it('should clone pre deployed contract with address', () => {
+			const contract = new Contract(
+				sampleStorageContractABI,
+				'0x00000000219ab540356cBB839Cbe05303d7705Fa',
+				{ gas: '0x97254' },
+			);
+
+			const clonnedContract = contract.clone();
+
+			expect(JSON.stringify(contract)).toStrictEqual(JSON.stringify(clonnedContract));
+		});
+
+		it('should clone new contract', () => {
+			const contract = new Contract(sampleStorageContractABI);
+
+			const clonnedContract = contract.clone();
+			expect(JSON.stringify(contract)).toStrictEqual(JSON.stringify(clonnedContract));
+		});
+
+		it('defaults set and get should work', () => {
+			const contract = new Contract([], '0x00000000219ab540356cBB839Cbe05303d7705Fa');
+
+			const defaultAddr = '0xd7E30ae310C1D1800F5B641Baa7af95b2e1FD98C';
+			expect(contract.defaultAccount).toBeUndefined();
+			contract.defaultAccount = defaultAddr;
+			expect(contract.defaultAccount).toStrictEqual(defaultAddr);
+
+			const defaultBlock = '0xC43A';
+			expect(contract.defaultBlock).toBe('latest');
+			contract.defaultBlock = defaultBlock;
+			expect(contract.defaultBlock).toStrictEqual(defaultBlock);
+
+			const defaultHardfork = 'constantinople';
+			expect(contract.defaultHardfork).toBe('london');
+			contract.defaultHardfork = defaultHardfork;
+			expect(contract.defaultHardfork).toStrictEqual(defaultHardfork);
+
+			const defaultCommon = {
+				customChain: { name: 'testnet', networkId: '5678', chainId: '5634' },
+				baseChain: 'mainnet',
+				hardFork: 'petersburg',
+			};
+			expect(contract.defaultCommon).toBeUndefined();
+			contract.defaultCommon = defaultCommon;
+			// expect(contract.defaultHardfork).toStrictEqual(defaultCommon);   // TODO check if is being fixed in 5419
```

Comment:
```suggestion
			// expect(contract.defaultCommon).toStrictEqual(defaultCommon);   // TODO check if is being fixed in 5419
```
Did u mean `defaultCommon` instead of `defaultHardFork`?

Reply:
yes. it was commented earlier, as this should not pass:
in earlier line I set HF diff as compared to contract.defaultCommon.hardfork
it should throw error when contract.defaultHardfork != contract.defaultCommon.hardfork we discussed in #5419 ?

Reply:
https://github.com/web3/web3.js/blob/4.x/packages/web3-eth/src/validation.ts#L161 It will error when theres a validation check against when common, chain and hardfork are all defined

Reply:
Thanks, so validation is done at `validateTransactionForSigning` level.

 I think we should have additional feature of doing validation during setting defaults as well, this will help users to identify errors at earlier stage.

Reply:
sounds good, that can be a different feature. pr looks good to me